gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/MostlyMySqlDagManagementStateStore.java

```diff
@@ -122,33 +125,36 @@ public Set<String> getFailedDagIds() throws IOException {
   }
 
   @Override
+  // todo - updating different mapps here and in addDagNodeState can result in inconsistency between the maps
   public synchronized void deleteDagNodeState(DagManager.DagId dagId, Dag.DagNode<JobExecutionPlan> dagNode) {
-    String dagIdStr = dagId.toString();
     this.jobToDag.remove(dagNode);
     this.dagNodes.remove(dagNode.getValue().getId());
-    this.dagToDeadline.remove(dagIdStr);
-    this.dagToJobs.get(dagIdStr).remove(dagNode);
-    if (this.dagToJobs.get(dagIdStr).isEmpty()) {
-      this.dagToJobs.remove(dagIdStr);
+    this.dagToDeadline.remove(dagId);
+    this.dagToJobs.get(dagId).remove(dagNode);
+    if (this.dagToJobs.get(dagId).isEmpty()) {
+      this.dagToJobs.remove(dagId);
     }
   }
 
+  // todo - updating different mapps here and in deleteDagNodeState can result in inconsistency between the maps
```

Comment:
I don't believe there's concern, since both methods are `synchronized`.  in fact, there's not need for CHM, since method-level synchronization